edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet/EdgeletCommandFactory.cs

```diff
@@ -12,12 +12,21 @@ public class EdgeletCommandFactory<T> : ICommandFactory
         readonly IConfigSource configSource;
         readonly IModuleManager moduleManager;
         readonly ICombinedConfigProvider<T> combinedConfigProvider;
+        readonly string edgeDeviceHostname;
+        readonly Option<string> parentEdgeHostname;
 
-        public EdgeletCommandFactory(IModuleManager moduleManager, IConfigSource configSource, ICombinedConfigProvider<T> combinedConfigProvider)
+        public EdgeletCommandFactory(
+            IModuleManager moduleManager,
+            IConfigSource configSource,
+            ICombinedConfigProvider<T> combinedConfigProvider,
+            string edgeDeviceHostname,
```

Comment:
One question - since the EdgeletCommandFactory takes in a IConfigSource, why do we need to pass in the edgeDeviceHostname and parentHostname?

Reply:
edgeDeviceHostname and parentHostname are passing from environment variables set by iotedged.  IConfigSource doesn't contain these information, e.g. TwinConfigSource contains all twin from IoT hub, but don't have these values.  And these values are used to determine what environment variables to be injected when create other modules.

Reply:
updated to use IConfigSource.Configuration.  we are peeking the implementation inside IConfigSource.Configuration, and this is an existing problem in multiple places in the solution.